src/main/java/com/netgrif/application/engine/pdf/generator/service/renderer/Renderer.java

```diff
@@ -57,4 +58,29 @@ protected static int getTextWidth(List<String> values, PDType0Font font, int fon
     protected int getMaxLabelLineSize(int fieldWidth, int fontSize) {
         return (int) ((fieldWidth - padding) * resource.getSizeMultiplier() / fontSize);
     }
+
+    public static String removeUnsupportedChars(String input, PdfResource resource) {
+        String value = Jsoup.parse(input.replaceAll("\\s{1,}", " ")).text();
+        value = Normalizer.normalize(value, Normalizer.Form.NFC);
+        StringBuilder b = new StringBuilder();
+        for (int i = 0; i < value.length(); i++) {
```

Comment:
is iterating over the characters in a string in this manner correct?

I have found information that indicates iterating like this might give wrong results for some characters (if they occupy two bytes)
https://stackoverflow.com/questions/1527856/how-can-i-iterate-through-the-unicode-codepoints-of-a-java-string

Reply:
2 bytes chars are not supported by PDF box currently, so that's why we are normalizing them. This iteration then gets only one byte chars.